Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/DriverPool.hs

```diff
@@ -441,8 +445,9 @@ calculateDriverPool ::
   Id DM.Merchant ->
   Bool ->
   Maybe PoolRadiusStep ->
+  [Id DP.Person] ->
   m [DriverPoolResult]
-calculateDriverPool poolStage driverPoolCfg mbVariant pickup merchantId onlyNotOnRide mRadiusStep = do
+calculateDriverPool poolStage driverPoolCfg mbVariant pickup merchantId onlyNotOnRide mRadiusStep blockListedDriverList = do
   let radius = getRadius mRadiusStep
```

Comment:
Fetch the block listed drivers directly here, and the filter here. Instead of passing them to Query or getting them from top.